### ISSUE TITLE
Proposal: <podcast:hasGuests>

### ISSUE THREAD

**Issue by jamescridland** [Podcastindex-org/podcast-namespace#378]
**What:**
A simple way to show, in the channel of a podcast RSS feed, whether a podcast has guests.

**How:**
`<podcast:hasGuests>yes</podcast:hasGuests>` - this podcast regularly has guests on its episodes

`<podcast:hasGuests>no</podcast:hasGuests>` - This podcast does **not** have guests on its episodes

`<podcast:hasGuests></podcast:hasGuests>` or an absence of this tag - This podcast may or may not have guests

**Why:**
* A method to ensure that promoters of guests do not waste time contacting podcasters with news of their "amazing guest who would be _perfect_ for your show"
* A method to signal to a UX element within a podcast player that this podcast has guests, and perhaps to display a different form of UX to allow individual guests to be clearly visible (using the already existing `podcast:person` tag)
* A method for podcast hosting companies to only ask about `podcast:person` for each show if that show has guests, and otherwise not surface that set of dialogs
* A method to populate podcast guesting tools such as Audry or Acast's Podchaser without being forced to claim shows and keep a separate set of databases updated
* A way to allow enhanced search within a podcast app and/or to signal to the app to perform a different filter on the description field or others to look for guest names and links; or to prepare for an unknown voice when producing a transcript; or other similar opportunities

**Impact:**
* Filesize: this is heavily compressible and would add negligible weight to the RSS feed
* This could be set, or unset, using a tickbox within the "podcast settings" section of a podcast host's UI, and should not add appreciably more time for the user
* This is a simple boolean value (yes/no/NULL) and has low impact on database schemas or storage

**Comment by daveajones** [comment]
I like the thinking here because just having a guest person tag doesn’t mean that you are a show that has guests. Some shows have guests infrequently, but it’s not part of their main format.

On the solicitation side, I wonder if that could be a different tag altogether. Something like `<podcast:solicit>` which describes the way you are willing to allow solicitation for things: guests, advertising, hosting, etc. 🤔

**Comment by theDanielJLewis** [comment]
I like this idea!

I also wonder if we might want to expand it to a bigger `<podcast:has>` tag that could potentially accept more types of "has …" indicators. I don't know what we might want to future-proof for, so it's only a consideration at this point.

**Comment by theDanielJLewis** [comment]
I like that thought, @HersheyCognosco. I only think that might be more appropriate to communicate on the podcaster's website as part of the "pitching" process.

**Comment by genebean** [comment]
What about renaming this to `acceptsGuests`? I think it would clarify the use better and be less likely to be confused with an episode that happens to have guests as represented by the guest bit of the person tag.

**Comment by jamescridland** [comment]
@genebean - I'm absolutely fine with that.

**Comment by theDanielJLewis** [comment]
Do you think it would be helpful to have more than a yes/no value? Like maybe some freeform text for the podcaster to say what kind of guests they want?

**Comment by jamescridland** [comment]
I'd really like to avoid freeform text. It's impossible for computers to understand.

**Comment by theDanielJLewis** [comment]
My thinking is that a "yes" value would signal to apps to display the freeform text within an appropriate context instead of indicating only "this show accepts guests."

**Comment by drpancake** [comment]
In case it's helpful in the discussion, we at Rephonic try to determine whether a podcast takes guests automatically (by looking for certain keywords and human names in episode metadata) and according to that measure 30% of all active shows take guests.

**Comment by genebean** [comment]
I’m with @jamescridland on the boolean aspect of the field so that it’s machine readable, but could certainly see the value of a sub tag that provided context to the human reading the results of the machine parsing. Another proposal I read yesterday did something similar.

**Comment by theDanielJLewis** [comment]
Yes, that's what I'm suggesting: an _additional_ attribute or style that can allow freeform text that can be displayed based on the boolean value.

**Comment by theDanielJLewis** [comment]
@si, that would be good to communicate and I think the right place for that would be in an additional attribute.

So here's my suggestion, in code:

```xml
<podcast:acceptsGuests description="We interview married entrepreneurs, and we love it when your spouse can join, too!">yes</podcast:acceptsGuests>
```

**Comment by si** [comment]
@theDanielJLewis good idea. That way, all contexts can be provided in the `description` attribute e.g. frequency, type of guests, any characteristics suitable.

**Comment by theDanielJLewis** [comment]
Oh, and probably a URL feed for more information or applying.

```xml
<podcast:acceptsGuests url="[URL for information or application]" description="We interview married entrepreneurs, and we love it when your 
  spouse can join, too!">yes</podcast:acceptsGuests>
```
